lib/src/utils.dart

```diff
@@ -48,17 +48,40 @@ abstract class ValueType {
   Value toValue();
 }
 
-// TODO fix To toValue() for List<T?>
-extension ListValueUtils<T extends ValueType> on List<T> {
+extension ListValueUtils<T> on List<T> {
   Value toValue() {
-    return Value(listValue: ListValue(values: map((e) => e.toValue())));
+    final values = map((e) {
+      try {
+        if (e is num) {
+          return Value(numberValue: e.toDouble());
+        } else if (e is String) {
+          return Value(stringValue: e);
+        } else if (e is bool) {
+          return Value(boolValue: e);
+        } else if (e is List<dynamic>) {
+          return e.toValue();
+        } else if (e is Map<String, dynamic>) {
+          return e.toValue();
+          // ignore: type_check_with_null, prefer_void_to_null
+        } else if (e is Null) {
+          return Value(nullValue: e);
+        } else {
+          throw GrpcError.invalidArgument('List contains unsupported type');
```

Comment:
I attempted to extract this logic into a helper function that both this and  Map<> to.Value() could use, because it is essentially copied from there, but I started running into the same issues we had before where it didn't find the correct toValue() at runtime, I may take a look at this again with a fresh mind.